pkg/datapath/linux/probes/probes.go

```diff
@@ -487,3 +434,171 @@ func HaveV3ISA() error {
 	}
 	return nil
 }
+
+// CreateHeaderFiles creates C header files with macros indicating which BPF
+// features are available in the kernel.
+func CreateHeaderFiles(headerDir string, probes *HeaderProbes) error {
+	common, err := os.Create(filepath.Join(headerDir, "features.h"))
+	if err != nil {
+		return fmt.Errorf("could not create common features headerfile: %w", err)
+	}
+	defer common.Close()
+	err = writeCommonHeader(common, probes)
+	if err != nil {
+		return fmt.Errorf("could not write common features headerfile: %w", err)
+	}
+
+	skb, err := os.Create(filepath.Join(headerDir, "features_skb.h"))
+	if err != nil {
+		return fmt.Errorf("could not create skb related features headerfile: %w", err)
+	}
+	defer skb.Close()
+	err = writeSkbHeader(skb, probes)
+	if err != nil {
+		return fmt.Errorf("could not write skb related features headerfile: %w", err)
+	}
+
+	xdp, err := os.Create(filepath.Join(headerDir, "features_xdp.h"))
+	if err != nil {
+		return fmt.Errorf("could not create xdp related features headerfile: %w", err)
+	}
+	defer xdp.Close()
+	err = writeXdpHeader(xdp, probes)
+	if err != nil {
+		return fmt.Errorf("could not write xdp related features headerfile: %w", err)
+	}
+
+	return nil
+}
+
+func ExecuteHeaderProbes() *HeaderProbes {
+	var probes HeaderProbes
+
+	mapProbeList := []ebpf.MapType{
+		ebpf.LRUHash,
+		ebpf.LPMTrie,
+	}
+	probes.Maps = make(map[ebpf.MapType]bool)
+	for _, mp := range mapProbeList {
+		if HaveMapType(mp) == nil {
+			probes.Maps[mp] = true
```

Comment:
If a feature is not supported, it will be omitted from `probes.Maps`, which might not be what we want. (this should be specified in the function doc)

To address this, remove the if and:
```suggestion
			probes.Maps[mp] = (HaveMapType(mp) == nil)
```